app/src/main/java/com/poupa/vinylmusicplayer/util/FileUtil.java

```diff
@@ -181,4 +187,68 @@ public static byte[] readBytes(InputStream stream) throws IOException {
         stream.close();
         return baos.toByteArray();
     }
+
+    public static File getSDCardDirectory(Context context) {
+        File sdFolder = null;
+        for (File dir : context.getExternalFilesDirs(null)) {
+            if(dir != null) {
+                if (!dir.equals(context.getExternalFilesDir(null))) {
+                    // first directory which is not primary storage - should be sd card
+                    String path = dir.getAbsolutePath();
+                    String base_path = path.substring(0, path.indexOf("Android/data"));
+                    sdFolder = new File(base_path);
+                    break;
+                }
+            }
+        }
+        return sdFolder;
+    }
+
+    /**
+     * Workaround to find all possible storage devices' root paths.
+     * s.a. '/sdcard/0', '/storage/emulated/0', or '/storage/SDCARD_NAME'.
+     *
+     * @return a non-null list of distinct Files, in no particular order.
+     */
+    public static List<File> getAllExternalStorageRootPaths(Context context){
```

Comment:
We could just use 

   - `Environment.getExternalStorageDirectory().getAbsolutePath()`; and 
   - `FileUtils.getSDCardDirectory` 

However, there may be a case in which an user has two different SD cards, or has their device connected to another form of secondary storage. 

 Since I can't test this scenario at the moment, I thought it would be best to give the user the greatest amount of freedom we can, while still checking for nulls and for `SecurityException`.